backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java

```diff
@@ -148,6 +143,21 @@ public Announcement releaseAnnouncement(ReleaseAnnouncementUseCase.Query query)
       allVersionsOfAllNormsToPublish.add(latestNormExpression);
     }
 
+    // Validate all resulting versions before making any actual changes to data in the system
+    allVersionsOfAllNormsToPublish.forEach(norm -> {
+      ldmlDeValidator.parseAndValidate(XmlMapper.toString(norm.getDocument()));
+      ldmlDeValidator.validateSchematron(norm);
+    });
+
+    // Delete the files from a previous release of the same announcement if they are still queued for publishing.
+    // This is to prevent residual files from remaining if some time boundaries changed and this release will create
+    // different expressions.
+    normsToPublish.forEach(norm ->
+      deleteQueuedNormsPort.deleteQueuedForPublishNorms(
+        new DeleteQueuedNormsPort.Command(norm.getWorkEli())
+      )
+    );
```

Comment:
this needs to happen in the beginning as otherwise we use wrong elis (see the comment).

It should not be a problem that this is modifying the database state as the whole method is annotated with `@Transactional` and will therefore only be executed if everything works out. Atleast thats my understanding, I haven't tested it.